sailor/assetcentral/system.py

```diff
@@ -162,8 +162,10 @@ def get_indicator_data(self, start: Union[str, pd.Timestamp, datetime.timestamp,
         end
             End of time series data.
         """
-        all_indicators = sum((equipment.find_equipment_indicators() for equipment in self.components), IndicatorSet([]))
-        return sap_iot.get_indicator_data(start, end, all_indicators, self.components)
+        if not hasattr(self, '_equipments'):
```

Comment:
I don't much like this bit (also below).
I guess the underlying problem is that `self._component_tree` secretly injects `self._equipments` into the class. 
When looking at this before I assumed `self._equipments` would only be used to build the component tree, but not as it's own property. 
Not sure what the best way of fixing this is though. @cloudyday, any suggestions?

Reply:
I don't see how it can be significantly improved without a major refactoring of the code. All the individual functions contributing to the final result have dependencies between each other using these _variables as state passing. It's okay since it's part of object oriented programming.